### PR TITLE
Allow TileFetchers to provide the coordinates for a tile.

### PR DESCRIPTION
**Pull request by ttung**
There are some TODOs introduced by this PR that @dganguli will probably need to fill in.

Test plan: `python examples/get_iss_breast_data.py /Users/ttung/Downloads/starfish_data/mignardi_breast_1/raw/ ../starfish-formatted-data/breast/` + visual inspection

Depends on #566

### CODE REVIEW

File: examples/get_iss_breast_data.py [hunk 1]
```diff
@@ -23,6 +23,15 @@ def __init__(self, file_path):
     def shape(self) -> Tuple[int, ...]:
         return SHAPE
 
+    @property
+    def coordinates(self) -> Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]:
```
Comment (ambrosejcarr) [inline examples/get_iss_breast_data.py]: Funny, Matt Cai mentioned today that coordinates are the one thing that's not in the filename when delivering single-plane TIFFs. Do we have a plan on how users should provide these by default? 

... a json file mapping filenames to coordinates, perhaps? It'll be important to set up such a default soon.

File: starfish/experiment/builder/providers.py [hunk 2]
```diff
@@ -25,6 +27,17 @@ def shape(self) -> Tuple[int, ...]:
...
         raise NotImplementedError()
 
+    @property
+    def coordinates(self) -> Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]:
+        """Return the tile's coordinates in the global coordinate space..
+
+        Returns
+        -------
+        Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]
+            Maps from a coordinate type to its value or range.
```
Comment (ambrosejcarr) [inline starfish/experiment/builder/providers.py]: I'd make this more expressive. `Maps from a coordinate type (e.g. 'x', 'y', or 'z') to its value or range.`

File: examples/get_iss_breast_data.py (hunk 1 above)
Comment (ttung) [reply examples/get_iss_breast_data.py]: > Do we have a plan on how users should provide these by default?

I don't envision that we'll be strongly opinionated about this process.

File: starfish/experiment/builder/providers.py (hunk 2 above)
Comment (ttung) [reply starfish/experiment/builder/providers.py]: Fixed.

**Comment by ttung** [comment]
> Is there a follow-up PR exposing coordinates in ImageStack and TileSet? :-)

Sort of.  Remember that coordinates are at the _Tile_ level, not TileSet.  Presumably one would have to index (using round/ch/z) to get at the coordinates.

**Comment by ambrosejcarr** [comment]
Ah, right. Perhaps registration methods could expose Coordinates on ImageStacks... we're going to need a way to shift IntensityTable coordinates when we concatenate them across adjacent tiles.

**Comment by ambrosejcarr** [comment]
I guess a follow-up question -- do we think users will ever give tiles in the same fov different coordinates? (should these _be_ fov level attributes?)

**Comment by ttung** [comment]
> I guess a follow-up question -- do we think users will ever give tiles in the same fov different coordinates? (should these be fov level attributes?)

yes, one of the data sets we have (Simone's, I think) has that characteristic.

**Pull request merged**